cdap-watchdog/src/main/java/co/cask/cdap/logging/save/KafkaMessageCallback.java

```diff
@@ -66,12 +71,17 @@ public void onReceived(Iterator<FetchedMessage> messages) {
 
     int count = 0;
 
+    long oldestProcessed = 0;
     while (messages.hasNext()) {
       FetchedMessage message = messages.next();
       try {
         GenericRecord genericRecord = serializer.toGenericRecord(message.getPayload());
         ILoggingEvent event = serializer.fromGenericRecord(genericRecord);
 
+        if (event.getTimeStamp() < oldestProcessed) {
```

Comment:
yes - it is timestamp. We also check for count - if we don't process any messages, we don't report